packages/protocol/contracts/L1/provers/GuardianProver.sol

```diff
@@ -153,16 +170,38 @@ contract GuardianProver is IVerifier, EssentialContract {
         nonReentrant
         returns (bool approved_)
     {
-        bytes32 hash = keccak256(abi.encode(_meta, _tran, _proof.data));
-        approved_ = _approve(_meta.id, hash);
+        bytes32 proofHash = keccak256(abi.encode(_meta, _tran, _proof.data));
+        uint256 _version = version;
+        bytes32 currProofHash = latestProofHash[_version][_meta.id];
 
-        emit GuardianApproval(msg.sender, _meta.id, _tran.blockHash, approved_, _proof.data);
+        if (currProofHash == 0) {
+            latestProofHash[_version][_meta.id] = proofHash;
+            currProofHash = proofHash;
+        }
+
+        bool conflicting = currProofHash != proofHash;
+        bool pauseProving =
+            conflicting && address(this) == resolve(LibStrings.B_CHAIN_WATCHDOG, true);
 
-        if (approved_) {
-            _deleteApproval(hash);
-            ITaikoL1(resolve(LibStrings.B_TAIKO, false)).proveBlock(
-                _meta.id, abi.encode(_meta, _tran, _proof)
-            );
+        if (conflicting) {
+            latestProofHash[_version][_meta.id] = proofHash;
+            emit ConflictingProofs(_meta.id, msg.sender, currProofHash, proofHash, pauseProving);
+        }
+
+        if (pauseProving) {
+            ITaikoL1(resolve(LibStrings.B_TAIKO, false)).pauseProving(true);
```

Comment:
I think we should also allow any guardian to directly pause as well because I think it will be useful to support all reasons to pause the bridge (for example too many guardian proofs necessary because proofs are broken). They basically already have the power to so by using (or abusing) this system so I think should be okay to do so.

Reply:
Can do. Please take a look at https://github.com/taikoxyz/taiko-mono/pull/17293

Reply:
If https://github.com/taikoxyz/taiko-mono/pull/17293 is good, I perfer to merge it later after this one is merged to main

Reply:
Done, take a look.